server/sonar-server/src/main/java/org/sonar/server/plugins/ws/InstalledPluginsWsAction.java

```diff
@@ -78,21 +94,61 @@ public void handle(Request request, Response response) throws Exception {
       );
   }
 
-  private void writeMetadataList(Response response, Collection<PluginMetadata> pluginMetadatas) {
-    JsonWriter jsonWriter = response.newJsonWriter();
-    jsonWriter.beginArray();
-    for (PluginMetadata pluginMetadata : pluginMetadatas) {
-      writeMetadata(jsonWriter, pluginMetadata);
+  private void writeMetadataList(JsonWriter jsonWriter, Collection<PluginMetadata> pluginMetadatas) {
+      jsonWriter.name(ARRAY_PLUGINS);
+      jsonWriter.beginArray();
+      for (PluginMetadata pluginMetadata : pluginMetadatas) {
+        writePluginMetadata(jsonWriter, pluginMetadata);
+      }
+      jsonWriter.endArray();
+  }
+
+  private void writePluginMetadata(JsonWriter jsonWriter, PluginMetadata pluginMetadata) {
+    jsonWriter.beginObject();
+
+    writeMetadata(jsonWriter, pluginMetadata);
+
+    writeOrganisation(jsonWriter, pluginMetadata);
+
+    writeUrls(jsonWriter, pluginMetadata);
+
+    writeArchive(jsonWriter, pluginMetadata);
+
+    jsonWriter.endObject();
+  }
+
+  private void writeArchive(JsonWriter jsonWriter, PluginMetadata pluginMetadata) {
+    if (pluginMetadata.getFile() == null) {
+      return;
     }
-    jsonWriter.endArray();
-    jsonWriter.close();
+
+    jsonWriter.name(OBJECT_ARCHIVE);
+    jsonWriter.beginObject();
+    jsonWriter.prop(PROPERTY_NAME, pluginMetadata.getFile().getName());
```

Comment:
> "archive" is there because the new WS is supposed to return more information that it currently can with the underlying components, see my email about that

this comment applied to the PR for WS listing pending installation and removal of plugins

nevertheless, the archive object will be used across all plugin Webservice but either under plugin or release